pkg/retriever/retriever.go

```diff
@@ -158,69 +161,55 @@ func (retriever *Retriever) isAcceptableStorageProvider(storageProviderId peer.I
 // isAcceptableQueryResponse determines whether a queryResponse is acceptable
 // according to the current configuration. For now this is just checking whether
 // PaidRetrievals is set and not accepting paid retrievals if so.
-func (retriever *Retriever) isAcceptableQueryResponse(peer peer.ID, queryResponse *retrievalmarket.QueryResponse) bool {
+func (retriever *Retriever) isAcceptableQueryResponse(peer peer.ID, req types.RetrievalRequest, queryResponse *retrievalmarket.QueryResponse) bool {
 	// filter out paid retrievals if necessary
-	return retriever.config.PaidRetrievals || totalCost(queryResponse).Equals(big.Zero())
+
+	acceptable := retriever.config.PaidRetrievals || totalCost(queryResponse).Equals(big.Zero())
+	if !acceptable {
+		log.Debugf("skipping query response from %s for %s: paid retrieval not allowed", peer, req.Cid)
+		retriever.spTracker.RemoveStorageProviderFromRetrieval(peer, req.RetrievalID)
```

Comment:
this is really nice that it can be pulled into this function because we have `req` now